web-app/src/common/prediction.ts

```diff
@@ -0,0 +1,24 @@
+import { GameState, StoneColor } from './gameState';
+import { Coordinates, deepCloneObject } from './utils';
+import { getNbMovesNeededToWin } from './pathfinding';
+
+export function getNextPlaySuggestion(gameState: GameState, stoneColor: StoneColor): Coordinates {
+  const allPotentialPlays = [];
+  gameState.board.forEach((row, y) => {
+    row.forEach((cell, x) => {
+      if (cell.value === "empty")
+        allPotentialPlays.push({ x, y })
+    })
+  });
+  if (allPotentialPlays.length === 0) throw new Error("There is no playable cell in the given board.")
+
+  const winCostForPotentialPlays = allPotentialPlays.map(playPosition => {
+    const potentialGameState = deepCloneObject(gameState) as GameState;
+    potentialGameState.board[playPosition.y][playPosition.x].value = stoneColor;
+    return { playPosition, cost: getNbMovesNeededToWin(potentialGameState, stoneColor) }
+  });
+
+  return winCostForPotentialPlays.reduce(function (prev, curr) {
+    return prev.cost < curr.cost ? prev : curr;

```

Comment:
I was wondering why your algo always returned cells closer to the right or bottom edge, but now I have my answer 😁 (`<` instead of `<=`). I find it a bit counter-intuitive for the IA to suggest moves that way, but it's not wrong either 😁 . So I guess I'm okay with this 😂

Reply:
I switch to give the priority to upper left cells !